snapcraft/cli/_errors.py

```diff
@@ -110,6 +108,57 @@ def exception_handler(exception_type, exception, exception_traceback, *,
     sys.exit(exit_code)
 
 
+def _is_send_to_sentry() -> bool:
+    # If ALWAYS has already been selected from before do not even bother to
+    # prompt again.
+    config_errors = None
+    try:
+        cli_config = _CLIConfig()
+        always_value = cli_config.get_option('Sentry', 'always_send')
+        if always_value == 'true':
+            return True
+    except errors.SnapcraftInvalidCLIConfigError as config_error:
+        raise config_error
+        echo.warning('There was an issue while trying to retrieve '
+                     'configuration data: {!s}'.format(config_error))
+        config_errors = config_error
+
+    # Either ALWAYS has not been selected in a previous run so resort to
+    # prompting.
+    response = _prompt_sentry()
+    if response in _YES_VALUES:
+        return True
+    elif response in _NO_VALUES:
+        return False
+    elif response in _ALWAYS_VALUES:
+        if config_errors is not None:
+            echo.warning(
+                'Not saving choice to always send data to Sentry as '
+                'the configuration file is corrupted.\n'
+                'Please edit and fix or alternatively remove the '
+                'configuration file {!r} from disk.'.format(
+                    config_errors.config_file))  # type: ignore
+        else:
+            click.echo('Saving choice to always send data to Sentry')
+            cli_config.set_option('Sentry', 'always_send', 'true')
+            cli_config.save()
+        return True
+    else:
+        echo.warning('Could not determine choice, assuming no.')
+        return False
+
+
+def _prompt_sentry():
+    msg = _MSG_SEND_TO_SENTRY_TRACEBACK_PROMPT
+    all_valid = _YES_VALUES + _NO_VALUES + _ALWAYS_VALUES
+
+    def validate(value):
+        if value.lower() in all_valid:
```

Comment:
We check that it's valid in lower-case, but then don't use `.lower()` when decided on the proper action. We probably should.

Reply:
interesting, they were there :-/

Reply:
Bah, I'm blind, sorry :grin: .